dev/benchmarks/macrobenchmarks/lib/src/stack_size.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi' as ffi;
+import 'dart:io' as io;
+
+import 'package:flutter/material.dart';
+
+import '../common.dart';
+
+typedef GetStackPointerCallback = int Function();
+
+// c interop function:
+// void* mmap(void* addr, size_t len, int prot, int flags, int fd, off_t offset);
+typedef CMmap = ffi.Pointer<ffi.Void> Function(
+    ffi.Pointer<ffi.Void>, ffi.IntPtr, ffi.Int32, ffi.Int32, ffi.Int32, ffi.IntPtr);
+typedef DartMmap = ffi.Pointer<ffi.Void> Function(
+    ffi.Pointer<ffi.Void>, int, int, int, int, int);
+final DartMmap mmap = ffi.DynamicLibrary.process().lookupFunction<CMmap, DartMmap>('mmap');
+
+// c interop function:
+// int mprotect(void* addr, size_t len, int prot);
+typedef CMprotect = ffi.Int32 Function(ffi.Pointer<ffi.Void>, ffi.IntPtr, ffi.Int32);
+typedef DartMprotect = int Function(ffi.Pointer<ffi.Void>, int, int);
+final DartMprotect mprotect = ffi.DynamicLibrary.process()
+    .lookupFunction<CMprotect, DartMprotect>('mprotect');
+
+const int protRead = 1;
+const int protWrite = 2;
+const int protExec = 4;
+
+const int mapPrivate = 0x02;
+const int mapJit = 0x0;
+const int mapAnon = 0x20;
+
+final GetStackPointerCallback getStackPointer = () {
+  if (!io.Platform.isAndroid) {
+    throw 'This benchmark test can only be run on Android.';
+  }
+  // Creates a block of memory to store the assembly code.
+  final ffi.Pointer<ffi.Void> region = mmap(ffi.nullptr, 4096, protRead | protWrite,
+      mapPrivate | mapAnon | mapJit, -1, 0);
+  if (region == ffi.nullptr) {
+    throw 'Failed to acquire memory for the test.';
+  }
+  // Writes the assembly code into the memory block. This assembly assuming we
+  // are running on arm64 devices, and it returns the memory address of the
+  // stack pointer.
+  region.cast<ffi.Uint8>().asTypedList(4096).setAll(
+      0,
+      <int>[
+        // "mov x0, sp"  in machine code: E0030091.
```

Comment:
Is there a reference page for arm64 assembly instructions you could link to here in the comments?

Reply:
I uses the online converter, I couldn't find a good link stating the machine code architecture. All I found are either random blog posts or pdf ebooks, I don't think i should reference them here. I intend to keep it as is.